### PR TITLE
Introduce a tkn bundle push command to produce Tekton bundles

### PR DESCRIPTION
**Pull request by pierretasci**
# Changes

Introduces a `tkn bundle push` command to produce a valid Tekton bundle.

This change is part of TEP#31: https://github.com/tektoncd/community/blob/main/teps/0031-tekton-bundles-cli.md

This adds a `bundle` top level command and a `bundle push` subcommand which takes file or `stdin` input (similar to `kubectl apply -f -`) to produce a Tekton bundle from a set of valid YAML or JSON Tekton objects.

It supports:
- multiple objects in one file
- stdin input
- fully-qualified Tekton Bundle references
- bearer token or basic auth

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
Adds a `tkn bundle push` command to generate valid Tekton Bundles
```

### CODE REVIEW

File: pkg/bundle/builder.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		}); err != nil {
+			return nil, err
+		}
+	}
+	return img, nil
+}
+
+// Return the ObjectMetadata.Name field which every resource should have.
+func getObjectName(obj runtime.Object) string {
+	return reflect.Indirect(reflect.ValueOf(obj)).FieldByName("ObjectMeta").FieldByName("Name").String()
```
Comment (vdemeester) [inline pkg/bundle/builder.go:80]: This line scares me 😹

**Review by vdemeester — COMMENTED** [review]
Question (on the design, without reviewing code) : 
- Is the `tkn push` additive or does it replace the full bundle ? as, can I incrementally (multiple commands) create my bundle, or should it be in a one-time operation ?

File: pkg/bundle/builder.go (hunk 1 above)
Comment (pierretasci) [reply pkg/bundle/builder.go:80]: You're right. I found a lib in apimachinery that does this for us so I will switch to using that.

**Comment by pierretasci** [comment]
> * Is the `tkn push` additive or does it replace the full bundle ? as, can I incrementally (multiple commands) create my bundle, or should it be in a one-time operation ?

The design called for a single command with no "builder" like features so this is a one-time operation which will replace bundles with the same name so long as you have write access to the repository.

**Review by vdemeester — APPROVED** [review]

**Comment by chmouel** [comment]
/lgtm

let's iterate from there, thanks

**Pull request merged**